### DISCUSSION TITLE
should the i18n packages be imported once globally, or in every component/server action where valibot is used

### DISCUSSION THREAD

**Discussion by stefanprobst** [open-circle/valibot#914] [Q&A] [answered]
should the i18n packages be imported once globally, or in every component/server action where valibot is used?

**Comment by fabian-hiller** [comment]
It depends. Have you read our i18n guide? https://valibot.dev/guides/internationalization/

**Reply by stefanprobst** [reply]
yep - i think what i am unclear about is what "global storage" and "sideeffect import" means in e.g. next.js - are server and client treated as two separate contexts with their separate "global storages", and therefore i need to ensure the i18n package is imported in both? or is this somehow handled by a bundler?

**Reply by fabian-hiller** [reply, marked as answer]
Ideally, you should only import the specific translations you need once, for example in the same file where you define your schemas. If you define your schemas on a per-route level, you should check that the translations are loaded correctly due to code splitting. There's no real harm in importing the translations more than once, so don't worry too much about it. But if you can only import them once, it's of course better. If a translation is imported twice, they will just overwrite each other.
